packages/chat-headless/src/ChatHeadless.ts

```diff
@@ -28,9 +30,29 @@ export class ChatHeadless {
   private chatCore: ChatCore;
   private stateManager: ReduxStateManager;
 
-  constructor(config: ChatConfig) {
+  /**
+   * Constructs a new instance of the {@link ChatHeadless} class.
+   *
+   * @public
+   *
+   * @param config - The configuration for the {@link ChatHeadless} instance
+   * @param saveToSessionStorage - Whether to save the instance's {@link ConversationState} to session storage
+   */
+  constructor(config: ChatConfig, saveToSessionStorage = true) {
     this.chatCore = new ChatCore(config);
     this.stateManager = new ReduxStateManager();
+    if (saveToSessionStorage) {
+      this.setState({
+        ...this.state,
+        conversation: loadSessionState(),
+      });
+      this.stateManager.getStore().subscribe(() => {
```

Comment:
you should use `this.addListener` instead:
```
this.addListener({
      valueAccessor: (s) => s.conversation,
      callback: () => sessionStorage.setItem(...),
    });
```
This way it will only update if ConversationState changes, not the whole entire state

Reply:
Ooh, nice! Done